modules/uhttp/client/client.go

```diff
@@ -25,19 +25,83 @@ import (
 	"time"
 
 	"go.uber.org/fx/auth"
+	"go.uber.org/fx/dig"
+	"go.uber.org/fx/modules"
+	"go.uber.org/fx/service"
+
+	"github.com/opentracing/opentracing-go"
+)
+
+const (
+	_middlewareKey = "httpClientMiddleware"
+	_graphKey      = "httpClientGraph"
 )
 
+// WithOutbound lets you to add a custom middleware to the client
+func WithOutbound(middleware ...OutboundMiddleware) modules.Option {
+	return func(info *service.ModuleCreateInfo) error {
+		items := info.Items
+		if m, ok := items[_middlewareKey]; ok {
+			items[_middlewareKey] = append(m.([]OutboundMiddleware), middleware...)
```

Comment:
I am consistent with current usage, I think we never check these: e.g. inboundMiddlewareFromCreateInfo, because this really means something went really wrong